File: frontend/src/lib/api/transactions.ts
import { API_URL, getAuthHeader } from '@/lib/api/client';

export interface TransactionRecord {
  id: string;
  ticker: string;
  action: 'buy' | 'sell';
  shares: number;
  price_per_share: number;
  currency: string;
  size_czk: number;
  realized_pnl_czk: number | null;
  recommendation_id: string | null;
  executed_at: string;
  notes: string | null;
}

export async function fetchTransactions(limit = 100): Promise<TransactionRecord[]> {
  const headers = await getAuthHeader();
  const response = await fetch(`${API_URL}/api/transactions/?limit=${limit}`, { headers });
  if (!response.ok) throw new Error('Nepodařilo se načíst transakce.');
  return response.json();
}

export interface ManualTradePayload {
  ticker: string;
  action: 'buy' | 'sell';
  shares: number;
  price_per_share: number;
  currency: string;
  executed_at: string;
  notes?: string | null;
  play_type: 'A' | 'B' | 'C';
}

export async function createManualTrade(payload: ManualTradePayload) {
  const headers = await getAuthHeader();
  const response = await fetch(`${API_URL}/api/transactions/manual`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      ...headers,
    },
    body: JSON.stringify(payload),
  });

  if (!response.ok) {
    const detail = await response.text();
    throw new Error(detail || 'Nepodařilo se uložit manuální obchod.');
  }

  return response.json();
}

File: backend/app/schemas/transactions.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class TransactionCreate(BaseModel):
    ticker: str
    action: str  # buy | sell
    shares: float
    price_per_share: float
    currency: str = "USD"
    executed_at: datetime
    recommendation_id: Optional[str] = None
    notes: Optional[str] = None


class TransactionResponse(BaseModel):
    id: str
    user_id: str
    ticker: str
    action: str
    shares: float
    price_per_share: float
    currency: str
    size_czk: Optional[float]
    recommendation_id: Optional[str]
    executed_at: datetime
    notes: Optional[str]
    created_at: datetime
